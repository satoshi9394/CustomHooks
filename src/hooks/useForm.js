import { useState } from 'react'

const useForm = (initialStatev = {}) => {

  const [ values, setValues ] = useState(initialStatev);

  const handleInpuntChange = ({target}) => {
    setValues({
      ...values,
      [ target.name ] : target.value
    });
  }

  return [ values, handleInpuntChange ]


}

export default useForm
