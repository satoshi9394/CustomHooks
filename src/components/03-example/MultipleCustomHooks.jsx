import React from 'react'
import { useFetch } from '../../hooks/useFetch';
import {useCounter} from '../../hooks/useCounter';




import './effectStyles.css';

const MultipleCustomHooks = () => {

  const { counter, increment } = useCounter(1)
  const url = `https://www.breakingbadapi.com/api/quotes/${counter}`

  const { loading, data } = useFetch(url)
  const { author, quote } = !!data && data[0];

  return (
    <div>
      <h1>
        BreakingBad Quetes
      </h1>
      <hr/>
      {
        loading ? 
        (
          <div
            className="alert alert-info text-center"
          >
          loading...
          </div>
        ) :
        (
          <blockquote
            className="blockquote text-right"
          >
            <p
              className="mb-0"
            >
              {quote}
            </p>
            <footer
              className="blockquote-footer"
            >
              {author}
            </footer>
          </blockquote>
        )
      }
      <button
        className="btn btn-primary"
        onClick={ () => {increment()} }
      >
        Siguiente capitulo
      </button>
    </div>
  )
}

export default MultipleCustomHooks
