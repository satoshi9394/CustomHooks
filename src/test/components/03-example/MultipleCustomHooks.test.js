import { shallow } from 'enzyme'
import React from 'react';
import MultipleCustomHooks from '../../../components/03-example/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas de <MultipleCustomHooks/>', () => {

  beforeEach( () => {
    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    })
  }) 

  test('debe de mostrarse correctamente', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    })
    const wrapper = shallow( <MultipleCustomHooks/>);
    expect( wrapper ).toMatchSnapshot();
  })

  test('debe de mostrar la informacion', () => {
    useFetch.mockReturnValue({
      data: [{
        author: 'Angel',
        quote: 'hola mundo'
      }],
      loading: false,
      error: null
    });
    const wrapper = shallow( <MultipleCustomHooks/>);
    expect( wrapper.find('.alert').exists() ).toBe(false);
    expect( wrapper.find('.mb-0').text().trim() ).toBe('hola mundo');
    expect( wrapper.find('footer').text().trim() ).toBe('Angel');
  })


})