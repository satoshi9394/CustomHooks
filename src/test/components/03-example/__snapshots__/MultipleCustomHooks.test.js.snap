// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <MultipleCustomHooks/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    BreakingBad Quetes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente capitulo
  </button>
</div>
`;
